Exiled.Events/Patches/Generic/StaminaUsage.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="StaminaUsage.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Generic
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features;
+    using HarmonyLib;
+    using NorthwoodLib.Pools;
+    using PlayerStatsSystem;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="StaminaStat.CurValue"/>.
+    /// </summary>
+    [HarmonyPatch(typeof(StaminaStat), nameof(StaminaStat.CurValue), MethodType.Setter)]
+    internal class StaminaUsage
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            Label returnLabel = generator.DefineLabel();
+            Label skipLabel = generator.DefineLabel();
+
+            newInstructions.InsertRange(0, new CodeInstruction[]
+            {
+                new(OpCodes.Ldfld, Field(typeof(StaminaStat), nameof(StaminaStat.Hub))),
```

Comment:
`StaminaStat.Hub` is not static, so you've to load its instance which in this case is represented by the `ldarg.0` opcode.